packages/runtime/src/modules/DeciderModule.ts

```diff
@@ -287,15 +295,35 @@ function filterConfigElementByPrefix(requestItemConfigElement: RequestItemDeriva
     return filteredRequestItemConfigElement;
 }
 
-function checkCompatibility(requestConfigElement: RequestConfig, requestOrRequestItem: LocalRequestDTO | RequestItemJSONDerivations): boolean {
+async function checkCompatibility(requestConfigElement: RequestConfig, requestOrRequestItem: LocalRequestDTO, services: RuntimeServices): Promise<boolean | Error>;
+async function checkCompatibility(requestConfigElement: RequestConfig, requestOrRequestItem: RequestItemJSONDerivations): Promise<boolean>;
+async function checkCompatibility(
```

Comment:
as this is not exported

```suggestion
async function checkCompatibility(
```

should be enough

Reply:
Given you suggestion, in line 442 of DeciderModule.ts the compiler doesn't know that checkCompatibility won't return an error.

Reply:
Also, it makes it clearer why we can use `services!` in line 319 in my opinion.

Reply:
Hm, I have to look at this in  the Editor.

Reply:
I still think it's over-engeneered for a helper method, but if it works - go :)